tests/test_duck_duck_go_search.py

```diff
@@ -23,5 +23,46 @@ def test_get_json(self):
 				# pp.pprint(data)
 				self.assertTrue(len(data.keys()) > 3)
 
+		@unittest.skipIf((os.getenv("API_KEY") == None), "no api_key provided")
+		def test_paginate_page_size(self):
```

Comment:
@dimitryzub, can you help me with tests based on pagination examples you've told?

Reply:
@ilyazub I've used the `pagination()` method which is worked without any workarounds.

Do we need to check for title equality as this is for test purposes to show if there're any duplicates? Or for something else?

https://github.com/serpapi/google-search-results-python/blob/bd65ec74a840f8fbe0b5f2c33512a3468009a5f6/tests/test_duck_duck_go_search.py#L56-L57

DuckDuckGo tends to display the same thing over multiple pages so there's could be a lot of duplicates. In my test where I paginate over all pages and extracted titles, the duplicate number was varying between 2 and 16.

____

`count % 2, 0` causes confusion to me. 

https://github.com/serpapi/google-search-results-python/blob/bd65ec74a840f8fbe0b5f2c33512a3468009a5f6/tests/test_duck_duck_go_search.py#L61

Sorry for not particularly useful comment.